RecoHI/HiJetAlgos/plugins/HiFJRhoFlowModulationProducer.cc

```diff
@@ -0,0 +1,251 @@
+// -*- C++ -*-
+//
+// Package:    RecoHI/HiJetAlgos/plugins/HiFJRhoFlowModulationProducer
+// Class:      HiFJRhoFlowModulationProducer
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/View.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/Provenance/interface/EventID.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/src/WorkerMaker.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h"
+#include "DataFormats/HeavyIonEvent/interface/EvtPlane.h"
+#include "DataFormats/JetReco/interface/JetCollection.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+#include "TF1.h"
+#include "TH1.h"
+#include "TMath.h"
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+class HiFJRhoFlowModulationProducer : public edm::stream::EDProducer<> {
+public:
+  explicit HiFJRhoFlowModulationProducer(const edm::ParameterSet&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginStream(edm::StreamID) override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endStream() override;
+
+  bool doEvtPlane_;
+  bool doFreePlaneFit_;
+  bool doJettyExclusion_;
+  int evtPlaneLevel_;
+  edm::EDGetTokenT<reco::JetView> jetTag_;
+  edm::EDGetTokenT<reco::PFCandidateCollection> pfCandsToken_;
+  edm::EDGetTokenT<reco::EvtPlaneCollection> evtPlaneToken_;
+
+  float* hiEvtPlane;
+};
+
+HiFJRhoFlowModulationProducer::HiFJRhoFlowModulationProducer(const edm::ParameterSet& iConfig)
+    : doEvtPlane_(iConfig.getParameter<bool>("doEvtPlane")),
+      doFreePlaneFit_(iConfig.getParameter<bool>("doFreePlaneFit")),
+      doJettyExclusion_(iConfig.getParameter<bool>("doJettyExclusion")),
+      evtPlaneLevel_(iConfig.getParameter<int>("evtPlaneLevel")),
+      jetTag_(consumes<reco::JetView>(iConfig.getParameter<edm::InputTag>("jetTag"))),
```

Comment:
If the event product pointed by `getTag_` is read only if `doJettyExclusion_ == true`, please consume it also only if `doJettyExclusion_ == true`. Otherwise a dependence is declared unnecessarily when `doJettyExclusion_ == false`.